src/components/Application/ProgressBar.tsx

```diff
@@ -68,95 +79,205 @@ const ProgressBar: React.FC<ProgressBarProps> = ({
     })
   }
 
+  // We want to be able to show FIVE states:
+  //    (but some may show same icon)
+  // error -> if at least one error or if not completed and strict
+  // success -> if completed and valid
+  // not started -> nothing has been filled in section/page
+  // incomplete -> started but not invalid (probably display same as not started)
+  // current -> "dot" in circle to show current section/page
+  const getIndicator = (
+    progress: ApplicationProgress,
+    isStrict: boolean,
+    isActive: boolean,
+    type = ProgressType.section
+  ) => {
+    const { completed, valid } = progress
+    const isStrictlylInvalid = !valid || (isStrict && !completed)
+    const isStarted = progress.doneNonRequired + progress.doneRequired > 0
+    const typeIndex = type === ProgressType.section ? 0 : 1
```

Comment:
nice, I like this approach

not a required change, but it's good to use descriptive object structure:
```
const progressIconMap = {
 error: {
    section: ...jsx
    page: ...jsx
  },
..
}
```